### ISSUE TITLE
Role task includes are broken in Ansible 2.0.1

### ISSUE THREAD

**Issue by Theaxiom** [ansible/ansible#14885]
##### Issue Type:

- Bug Report
##### Ansible Version:

```
ansible 2.0.1.0
  config file =
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

Default
##### Environment:

CentOS 7
3.10.0-327.10.1.el7.x86_64
##### Summary:

Includes appear to be silently crashing the current role execution.
##### Steps To Reproduce:

site.yml

``` site.yml

---
- name: Ansible test Run
  hosts: localhost
  connection: local
  become: True
  pre_tasks:
  roles:
    - test
```

localhost

``` localhost
127.0.0.1 ansible_connection=local
```

roles/test/tasks/main.yml

``` roles/test/tasks/main.yml

---
- name: "Test 1"
  debug: msg="Hello World"

- include: include.yml

- name: "Test 3"
  debug: msg="Hello World"
```

roles/test/tasks/include.yml

``` roles/test/tasks/include.yml

---
- name: "Test 2"
  debug: msg="Hello World"
```

##### Expected Results:

```
[vagrant@localhost ansible-test]$ cat localhost
127.0.0.1 ansible_connection=local
[vagrant@localhost ansible-test]$ ansible-playbook site.yml -i localhost -vvvv
No config file found; using defaults
Loaded callback default of type stdout, v2.0

PLAYBOOK: site.yml *************************************************************
1 plays in site.yml

PLAY [Ansible HostType test Run] ***********************************************

TASK [setup] *******************************************************************
ESTABLISH LOCAL CONNECTION FOR USER: vagrant
127.0.0.1 EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558 `" )'
127.0.0.1 PUT /tmp/tmpdymSfX TO /home/vagrant/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558/setup
127.0.0.1 EXEC /bin/sh -c 'sudo -H -S -n -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-xugffqsjfkddnitukzzqsulxllqkbmbc; LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558/setup; rm -rf "/home/vagrant/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558/" > /dev/null 2>&1'"'"''
ok: [127.0.0.1]

TASK [test : Test 1] ***********************************************************
task path: /home/vagrant/ansible-test/roles/test/tasks/main.yml:2
ok: [127.0.0.1] => {
    "failed": false,
    "msg": "Hello World"
}

TASK [test : include] **********************************************************
task path: /home/vagrant/ansible-test/roles/test/tasks/main.yml:5
included: /home/vagrant/ansible-test/roles/test/tasks/include.yml for 127.0.0.1

TASK [test : Test 2] ***********************************************************
task path: /home/vagrant/ansible-test/roles/test/tasks/include.yml:2
ok: [127.0.0.1] => {
    "failed": false,
    "msg": "Hello World"
}

TASK [test : Test 3] ***********************************************************
task path: /home/vagrant/ansible-test/roles/test/tasks/main.yml:7
ok: [127.0.0.1] => {
    "failed": false,
    "msg": "Hello World"
}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=3    changed=0    unreachable=0    failed=0
```
##### Actual Results:

```
[vagrant@localhost ansible-test]$ cat localhost
127.0.0.1 ansible_connection=local
[vagrant@localhost ansible-test]$ ansible-playbook site.yml -i localhost -vvvv
No config file found; using defaults
Loaded callback default of type stdout, v2.0

PLAYBOOK: site.yml *************************************************************
1 plays in site.yml

PLAY [Ansible HostType test Run] ***********************************************

TASK [setup] *******************************************************************
ESTABLISH LOCAL CONNECTION FOR USER: vagrant
127.0.0.1 EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558 `" )'
127.0.0.1 PUT /tmp/tmpdymSfX TO /home/vagrant/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558/setup
127.0.0.1 EXEC /bin/sh -c 'sudo -H -S -n -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-xugffqsjfkddnitukzzqsulxllqkbmbc; LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558/setup; rm -rf "/home/vagrant/.ansible/tmp/ansible-tmp-1457559860.3-140581205680558/" > /dev/null 2>&1'"'"''
ok: [127.0.0.1]

TASK [test : Test 1] ***********************************************************
task path: /home/vagrant/ansible-test/roles/test/tasks/main.yml:2
ok: [127.0.0.1] => {
    "failed": false,
    "msg": "Hello World"
}

TASK [test : include] **********************************************************
task path: /home/vagrant/ansible-test/roles/test/tasks/main.yml:5
included: /home/vagrant/ansible-test/roles/test/tasks/include.yml for 127.0.0.1

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=3    changed=0    unreachable=0    failed=0
```

**Comment by jimi-c** [comment]
Hi @Theaxiom, I believe I may have fixed this last week or so when fixing other bugs. Running your example in devel today gives the following output:

<pre>
Using /etc/ansible/ansible.cfg as config file
PLAYBOOK: test.yml *************************************************************
1 plays in test.yml
PLAY [Ansible test Run] ********************************************************
TASK [setup] *******************************************************************
ok: [localhost]
TASK [test : Test 1] ***********************************************************
task path: /root/testing/14885/roles/test/tasks/main.yml:3
ok: [localhost] => {
    "msg": "Hello World"
}
TASK [test : include] **********************************************************
task path: /root/testing/14885/roles/test/tasks/main.yml:6
included: /root/testing/14885/roles/test/tasks/include.yml for localhost
TASK [test : Test 2] ***********************************************************
task path: /root/testing/14885/roles/test/tasks/include.yml:2
ok: [localhost] => {
    "msg": "Hello World"
}
TASK [test : Test 3] ***********************************************************
task path: /root/testing/14885/roles/test/tasks/main.yml:8
ok: [localhost] => {
    "msg": "Hello World"
}
PLAY RECAP *********************************************************************
localhost                  : ok=5    changed=0    unreachable=0    failed=0   
</pre>

I have not cherry-picked over my fixes yet, however this will be done as we gear up to release 2.0.2.

Can you test on devel if possible and see if you continue to see any issues related to this?

**Comment by Theaxiom** [comment]
Sorry I have been extremely busy and just saw this, I will try this out when I get home this evening.

**Comment by jimi-c** [comment]
Hi @Theaxiom, as there's been no follow-up in quite a while and we're seeing this working, I'm going to go ahead and close this now. If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by jimi-c** [comment]
Going to go ahead and close this, as the fix as been in devel and stable-2.0 for a few weeks now. These fixes will be included in 2.0.2 and 2.1.0.

If  you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by Theaxiom** [comment]
Thanks! I'm sorry I've been extremely swamped the last month and have been traveling for work. I really appreciate the hard work. :)
